include/deal.II/base/parameter_handler.h

```diff
@@ -846,57 +846,113 @@ class ParameterHandler : public Subscriptor
 {
 public:
   /**
-   * List of possible output formats used for
-   * ParameterHandler::print_parameters().
+   * List of possible output formats used for functions like
+   * ParameterHandler::print_parameters(). The options can be categorized into
+   * two groups:
+   * - format options: PRM, LaTeX, Description, XML, JSON
+   * - stylistic options: Short, KeepDeclarationOrder
+   *
+   * A valid combination of options is given in the following table:
+   *
+   * Stylistic options    | Default | PRM | LaTeX | Description | XML | JSON
+   * -------------------- | ------- | --- | ----- | ----------- | --- | ----
+   * Default              |    X    |  X  |   X   |      X      |  X  |  X
+   * Short                |    X    |  X  |   -   |      X      |  X  |  X
+   * KeepDeclarationOrder |    X    |  X  |   X   |      X      |  X  |  X
+   *
+   * Valid combinations are checked at the beginning of the functions that
+   * accept an OutputStyle as an input argument.
+   *
+   * Furthermore, a number of shortcuts of commonly used option combinations are
+   * provided. E.g., ShortPRM prints the parameters in the PRM
+   * format, while skipping the documentation.
    */
   enum OutputStyle
   {
     /**
-     * Write human readable output suitable to be read by ParameterHandler
-     * again.
+     * Default stylistic style: print documentation and sort all parameters
+     * alphabetically.
+     */
+    DefaultStyle = 0x0000,
```

Comment:
I regret that we hadn't settled on `use_underscore` instead of `CamelCase` for member variables and enum values back when this class was written :-(

But it's too late now. It makes more sense to keep the members of this class consistent with each other than to have some kind of mixture.

Reply:
@bangerth We can still deprecate all old names. In that case, the question would be how to call `Short` since `short` is already taken.

Reply:
Let's discuss this in another PR...

Reply:
Yes. I don't think there is much of an upside to making such a change. It is what it is.